cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -246,6 +246,7 @@ public KafkaReconciler(
      */
     public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
         return modelWarnings(kafkaStatus)
+                .compose(i -> PreventBrokerScaleDownUtils.canScaleDownBrokers(vertx, reconciliation, kafka, kafkaStatus, secretOperator, adminClientProvider, currentReplicas))
```

Comment:
I guess we can't do that because other replicas count is being used in multiple places before the `scaleDown()` method. So to avoid any sort of partial scale down this is the best place